pulsar-client-cpp/lib/Backoff.cc

```diff
@@ -21,20 +21,34 @@
 
 namespace pulsar {
 
-Backoff::Backoff(const TimeDuration& initial, const TimeDuration& max)
+Backoff::Backoff(const TimeDuration& initial, const TimeDuration& max, const TimeDuration& mandatoryStop)
         : initial_(initial),
           max_(max),
-          next_(initial) {
+          next_(initial),
+          mandatoryStopMade_(false),
+          mandatoryStop_(mandatoryStop) {
 }
 
 TimeDuration Backoff::next() {
     TimeDuration current = next_;
     next_ = std::min(next_ * 2, max_);
+
+    static TimeDuration timeElapsedSinceDisconnection_;
```

Comment:
This is not working: 
 1. Backoff has no notion of disconnection.. it's just a simple class to deal with exponential backoff
 2. `static` here is messing up across different `Backoff` instances. Each connection is having its own instance.

Reply:
Rookie Mistake :-(